### PR TITLE
Allow for .precious.toml as an alternate default config file name

### LINKED ISSUE #13: Possibly allow for arbitrary config file name
I introduced a `precious.toml` file internally at $work and there was a suggestion to use `.precious.toml` in order to hide the file from the people who aren't yet using the tool. It would be nice to allow for an arbitrary config file name to be supplied or at least for a config file to have a leading dot.

I could work on this if it's something you're amenable to.

### PR DESCRIPTION
**Pull request by oalders**
- Allow for .precious.toml as a default config file name
- Use a constant for precious.toml in tests

Fixes #13

### CODE REVIEW

**Comment by oalders** [comment]
This is my first ever Rust code, so I'm open to lots of harsh feedback. 📣

**Review by autarch — CHANGES_REQUESTED** [review]
Just a few small comments. Overall this looks good, but it also needs doc updates (just the `README.md`).

File: src/precious.rs [hunk 1]
```diff
@@ -680,8 +684,13 @@ impl<'a> Precious<'a> {
...
     fn has_config_file(path: &Path) -> bool {
         let mut file = path.to_path_buf();
+        file.push(".precious.toml");
+        if file.exists() {
+            return true;
+        }
+        file.pop();
         file.push("precious.toml");
-        file.exists()
+        return file.exists();
```
[the remark this answers is not in the document]
Comment (oalders) [reply src/precious.rs:693]: In that case I get

>  1 src/precious.rs|688 col 8-22 error| mismatched types: implicitly returns `()` as its body has no tail or `return` expression
>  2 src/precious.rs|688 col 40-43 error| mismatched types: expected `bool`, found `()`

File: src/precious.rs [hunk 2]
```diff
@@ -305,7 +305,11 @@ impl<'a> Precious<'a> {
 
     fn default_config_file(root: &Path) -> PathBuf {
         let mut file = root.to_path_buf();
-        file.push("precious.toml");
+        file.push(".precious.toml");
+        if !file.exists() {
```
[the remark this answers is not in the document]
Comment (oalders) [reply src/precious.rs:309]: I ended up with something like:

```rust
    fn default_config_file(root: &Path) -> PathBuf {
        let root_dir = root.to_path_buf();
        let mut file;
        for name in ["precious.toml", ".precious.toml"] {
            file = root_dir.clone();
            file.push(name);
            if file.exists() {
                return file;
            }
        }
        file
    }
```

That leaves me with 

> src/precious.rs|316 col 9-12 error| use of possibly-uninitialized variable: `file`: use of possibly-uninitialized `file`

**Comment by oalders** [comment]
I'm ready for my next round of feedback. 😄

File: src/precious.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (autarch) [reply src/precious.rs]: I pushed a change to do this with an iterator using `itertools` (Rust's `List::SomeUtils`), but then I realized that'd bump the itertools dependency version up, and I had tried to lower that version for the benefit of Debian, so I'm going to do the same thing with an iterator manually.

**Comment by autarch** [comment]
I ended up merging my version from the CLI. I would have tried to give you feedback so you could do it yourself, but the only way I could get the logic I wanted to work was to actually implement and test it.

This will be in the next release.

**Comment by oalders** [comment]
Great. Thanks for doing that! Was a good learning experience for me.

**Pull request closed without merging**